src/common/latest/helpers/logging.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import * as tl from "azure-pipelines-task-lib/task";
+import { EndpointType } from "../sonarqube";
+
+export enum LogLevel {
+  DEBUG = "DEBUG",
+  INFO = "INFO",
+  WARN = "WARN",
+  ERROR = "ERROR",
+}
+
+const LOG_MESSAGE_PADDING = 7;
+
+let endpointType: EndpointType | undefined;
+
+export function log(level: LogLevel, message: string) {
+  const levelStr = `[${level}]`.padEnd(LOG_MESSAGE_PADDING);
+  const msgStr = [levelStr, `${endpointType ?? EndpointType.SonarQube}:`, message].join(" ");
+  switch (level) {
+    case LogLevel.ERROR:
+      tl.error(msgStr);
+      break;
+    case LogLevel.WARN:
+      tl.warning(msgStr);
+      break;
+    case LogLevel.INFO:
+      console.log(msgStr);
```

Comment:
looking at the validate-ci task where the tests are run, I am seeing duplicate for each console.log

https://dev.azure.com/sonarsource/DotNetTeam%20Project/_build/results?buildId=100618&view=logs&j=f1338668-1e5b-508a-0652-45679475677e&t=41df0c21-6a86-5c69-c5a6-7f399df43a84&l=518

Reply:
I don't think they're duplicate, they're different (similar) tests 🤔

Reply:
yeah, I think you're right. Usually I would prefer no logs at all in tests. If we care to know they're called, we can mock and assert it was called with the expected message. I think you did this for debug and error

```
    jest.mocked(tl.error).mockImplementation(() => {});
    jest.mocked(tl.debug).mockImplementation(() => {});
```

maybe we do the same for console.log since it's now the INFO level logging? WDYT?